src/Microsoft.Extensions.WebSockets.Internal/WebSocketConnection.cs

```diff
@@ -249,8 +247,18 @@ private void WriteMaskingKey(Span<byte> buffer)
             _maskingKeyBuffer.CopyTo(buffer);
         }
 
+        /// <summary>
+        /// Terminates the socket abruptly.
+        /// </summary>
+        public void Abort()
+        {
+            _terminateReceiveCts.Cancel();
```

Comment:
`CancelPendingRead` just releases the `ReadAsync` call, we need to cover the cancellation case where we aren't in a `ReadAsync` (L392 below)